storm-client/src/jvm/org/apache/storm/utils/Utils.java

```diff
@@ -122,7 +122,9 @@
     // tests by subclassing.
     private static Utils _instance = new Utils();
     private static String memoizedLocalHostnameString = null;
-    public static final Pattern TOPOLOGY_KEY_PATTERN = Pattern.compile("^[\\w \\t\\._-]+$", Pattern.UNICODE_CHARACTER_CLASS);
+    public static final Pattern BLOB_KEY_PATTERN =
+            Pattern.compile("^[\\w \\t\\._-]+$", Pattern.UNICODE_CHARACTER_CLASS);
+    private static final Pattern TOPOLOGY_NAME_REGEX = Pattern.compile("^[^/.:\\\\]+$");
```

Comment:
Copied `TOPOLOGY_NAME_REGEX` from storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java. We should have a consistent check for topology name. 

`TOPOLOGY_KEY_PATTERN` was originally only used for blob key. See https://github.com/apache/storm/pull/2703